runtime/vm/BytecodeInterpreter.hpp

```diff
@@ -5278,6 +5287,40 @@ class INTERPRETER_CLASS
 			} else if (J9NtcPointer == argType) {
 				/* ffi_call expects the address of the pointer is the address of the stackslot. */
 				pointerValues[i] = (U_64)ffiArgs[i];
+#if JAVA_SPEC_VERSION >= 22
+				if (J9_ARE_ALL_BITS_SET(pointerValues[i], J9_FFI_DOWNCALL_HEAP_ARGUMENT_ID)) {
```

Comment:
Shouldn't this compare the whole value rather than just one bit?
```c
				if ((U_64)J9_FFI_DOWNCALL_HEAP_ARGUMENT_ID == pointerValues[i]) {
```

Reply:
Agreed & Fixed as suggested above.